fdbserver/ClusterController.actor.cpp

```diff
@@ -1692,20 +1692,37 @@ class ClusterControllerData {
 		if (req.configuration.regions.size() > 1) {
 			std::vector<RegionInfo> regions = req.configuration.regions;
 			if (regions[0].priority == regions[1].priority && regions[1].dcId == clusterControllerDcId.get()) {
+				TraceEvent("CCSwitchPrimaryDC", id)
+				    .detail("CCDcId", clusterControllerDcId.get())
+				    .detail("OldPrimaryDCId", regions[0].dcId)
+				    .detail("NewPrimaryDCid", regions[1].dcId);
 				std::swap(regions[0], regions[1]);
 			}
 
 			if (regions[1].dcId == clusterControllerDcId.get() &&
 			    (!versionDifferenceUpdated || datacenterVersionDifference >= SERVER_KNOBS->MAX_VERSION_DIFFERENCE)) {
 				if (regions[1].priority >= 0) {
+					TraceEvent("CCSwitchPrimaryDC", id)
```

Comment:
One thing that may be worth thinking about is whether it will be straightforward to distinguish between the two cases above that both log `CCSwitchPrimaryDC`. Sometimes when we have events with the same name and no distinguishing characteristics, an investigation will leave you wondering which of the two branches you took, and in those cases it's useful to make the trace events different in some way.

I'm not sure if that applies here, so I'll let you decide what makes sense.

Reply:
Thanks!